src/d/actor/d_a_wall.cpp

```diff
@@ -4,99 +4,354 @@
     /* Nonmatching */
+    typedef void (daWall_c::*procFunc)();
+    static procFunc mode_proc[] = {
+        &daWall_c::mode_wait,
+        &daWall_c::mode_break,
+    };
+
+    (this->*mode_proc[mState])();
+    return 1;
 }
 
 /* 00000980-00000A58       .text mode_wait__8daWall_cFv */
 void daWall_c::mode_wait() {
-    /* Nonmatching */
+    for (int i = 0; i < 2; i++) {
+        if (mTri[i].ChkTgHit()) {
+            cCcD_Obj * at = mTri[i].GetTgHitObj();
+            if (at != NULL && at->ChkAtType(AT_TYPE_BOMB)) {
+                set_effect();
+                set_se();
+                break;
+            } else {
+                mTri[i].ClrTgHit();
+            }
+        }
+    }
+
+    for (int i = 0; i < 2; i++) {
+        dComIfG_Ccsp()->Set(&mTri[i]);
+    }
 }
 
 /* 00000A58-00000B14       .text mode_break__8daWall_cFv */
 void daWall_c::mode_break() {
-    /* Nonmatching */
+    if (mBreakCounter != 0) {
+        mBreakCounter = mBreakCounter + 1;
+        switch (mType) {
+        case 0:
+        case 1:
+        case 2:
+            int tmp;
+            if ((mBreakCounter > 10) && (tmp = cLib_chaseF(&mpDst, 0.0, 2.222222), tmp != 0)) {
+                fopAcM_delete(this);
+            }
+
+            JPABaseEmitter* pEmitter = mSmokeCb.mpEmitter;
+            if (pEmitter != NULL) {
+                pEmitter->mGlobalPrmColor.a = mpDst;
+            }
+            break;
+        default:
+            fopAcM_delete(this);
+            break;
+        }
+    }
+    fopDwTg_DrawQTo(&mDwTg);
 }
 
 /* 00000B14-00000D84       .text set_tri__8daWall_cFv */
 void daWall_c::set_tri() {
-    /* Nonmatching */
+    for (int i = 0; i < 2; i++) {
+        mTri[i].Set(l_tri_src);
+        mTri[i].SetStts(&mStts);
+    }
+
+    Vec vertex[3];
+    vertex[0] = m_tri_vtx[mType][0];
+    vertex[1] = m_tri_vtx[mType][1];
+    vertex[2] = m_tri_vtx[mType][2];
+    mDoMtx_stack_c::transS(fopAcM_GetPosition(this));
+    mDoMtx_stack_c::YrotM(fopAcM_GetAngle_p(this)->y);
+
+    for (int i = 0; i < 3; i++) {
+        mDoMtx_stack_c::multVec(&vertex[i], &vertex[i]);
+    }
+    mTri[0].mA = vertex[0];
+    mTri[0].mB = vertex[1];
+    mTri[0].mC = vertex[2];
+
+    cM3d_CalcPla(&mTri[0].mA, &mTri[0].mB, &mTri[0].mC, &mTri[0].mNormal, &mTri[0].mD);
+
+    vertex[0] = m_tri_vtx[mType][0];
+    vertex[1] = m_tri_vtx[mType][2];
+    vertex[2] = m_tri_vtx[mType][3];
+    for (int i = 0; i < 3; i++) {
+        mDoMtx_stack_c::multVec(&vertex[i], &vertex[i]);
+    }
+    mTri[1].mA = vertex[0];
+    mTri[1].mB = vertex[1];
+    mTri[1].mC = vertex[2];
+    cM3d_CalcPla(&mTri[1].mA, &mTri[1].mB, &mTri[1].mC, &mTri[1].mNormal, &mTri[1].mD);
 }
 
 /* 00000D84-00000F74       .text set_effect__8daWall_cFv */
 void daWall_c::set_effect() {
-    /* Nonmatching */
+    u16 projection_id[3] = {0xa16e, 0xa170, 0xa172};
+    u16 particle_id[3] = {0xa16f, 0xa171, 0xa173};
+
+    csXyz local_28 = current.angle;
+    local_28.y += 0x8000;
+    mpDst = 200.00;
```

Comment:
```suggestion
    mDst = 200.0f;
```
this is a single not a double so it should have `f`